src/visualizers/widgets/MonacoEditor/MonacoEditorWidget.js

```diff
@@ -0,0 +1,476 @@
+                    } else {
+                        let themeName = name;
+
+                        if (isSet) {
+                            themeName = theme.toLowerCase();
+                        }
+
+                        await this.importTheme(themeName, ThemeList[theme]);
+                    }
+                    this.displaySettings.theme = theme;
+                    this.onUpdateDisplaySettings();
+                },
+                className: 'vs-dark'
+            };
+        });
+
+        AVAILABLE_KEYBINDINGS.forEach(name => {
+            const handler = name.toLowerCase().replace(/ /g, '_'),
+                isSet = handler === this.displaySettings.keybindings;
+
+            if (isSet) {
+                name = '<span style="font-weight: bold">' + name + '</span>';
+            }
+
+            menuItems.setKeybindings.items[handler] = {
+                name: name,
+                isHtmlName: isSet,
+                callback: () => {
+                    this.activateKeyBinding(handler);
+                    this.displaySettings.keybindings = handler;
+                    this.onUpdateDisplaySettings();
+                },
+                className: 'vs-dark',
+            };
+        });
+
+        return menuItems;
+    };
+
+    MonacoEditorWidget.prototype.setNavColor = function (color) {
+        const $tabNav = this._el.find('.nav-display-menu');
+        $tabNav.css({
+            'background-color': ''
+        });
+        $tabNav.css({
+            'background-color': color
+        });
+        this.displaySettings.navColor = color;
+    };
+
+    MonacoEditorWidget.prototype.importTheme = async function (name, theme) {
+        if (!(name && theme)) {
+            return;
+        }
+        const themeURL = `text!MonacoThemes/${theme}.json`;
+        const self = this;
+        return new Promise((resolve, reject) => {
+            require([themeURL], function (themeJSON) {
+                themeJSON = JSON.parse(themeJSON);
+                monaco.editor.defineTheme(name, themeJSON);
+                monaco.editor.setTheme(name);
+                if (themeJSON.colors['editor.background']) {
+                    self.setNavColor(themeJSON.colors['editor.background']);
+                }
+                resolve();
+            }, reject);
+        });
+    };
+
+    MonacoEditorWidget.prototype.activateKeyBinding = function (keybinding) {
+        if (keybinding === 'default' && this.displaySettings.keybindings === 'vim') {
+            this.disposeVimMode();
+        } else if (keybinding === 'vim') {
+            this.initVimMode();
+        }
+    };
+
+    MonacoEditorWidget.prototype.initVimMode = function () {
+        const self = this;
+        return new Promise((resolve, reject) => {
+            require(['MonacoVim'], function (MonacoVim) {
+                self.keyBindingAdapter = MonacoVim.initVimMode(self.editor, self.$status[0]);
```

Comment:
It would be good to use consistent capitalization: `keybindingAdapter` (to be consistent with `this.displaySettings.keybindings`)